spring-integration-hazelcast/src/main/java/org/springframework/integration/hazelcast/leader/LeaderInitiator.java

```diff
@@ -0,0 +1,253 @@
+import org.springframework.integration.leader.DefaultCandidate;
+import org.springframework.integration.leader.event.DefaultLeaderEventPublisher;
+import org.springframework.integration.leader.event.LeaderEventPublisher;
+import org.springframework.util.Assert;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+
+/**
+ * Bootstrap leadership {@link org.springframework.cloud.cluster.leader.Candidate candidates}
+ * with Hazelcast. Upon construction, {@link #start} must be invoked to
+ * register the candidate for leadership election.
+ *
+ * @author Patrick Peralta
+ * @author Gary Russell
+ */
+public class LeaderInitiator implements Lifecycle, InitializingBean, DisposableBean, ApplicationEventPublisherAware {
+
+	private static final Logger logger = LoggerFactory.getLogger(LeaderInitiator.class);
+
+	/**
+	 * Hazelcast client.
+	 */
+	private final HazelcastInstance client;
+
+	/**
+	 * Candidate for leader election.
+	 */
+	private final Candidate candidate;
+
+	/**
+	 * Executor service for running leadership daemon.
+	 */
+	private final ExecutorService executorService = Executors.newSingleThreadExecutor(new ThreadFactory() {
+		@Override
+		public Thread newThread(Runnable r) {
+			Thread thread = new Thread(r, "Hazelcast leadership");
+			thread.setDaemon(true);
+			return thread;
+		}
+	});
+
+	/**
+	 * Future returned by submitting an {@link Initiator} to {@link #executorService}.
+	 * This is used to cancel leadership.
+	 */
+	private volatile Future<Void> future;
+
+	/**
+	 * Hazelcast distributed map used for locks.
+	 */
+	private volatile IMap<String, String> mapLocks;
+
+	/**
+	 * Flag that indicates whether the leadership election for
+	 * this {@link #candidate} is running.
+	 */
+	private volatile boolean running;
+
+	/**
+	 * Leader event publisher.
+	 */
+	private volatile LeaderEventPublisher leaderEventPublisher = new DefaultLeaderEventPublisher();
+
+	private boolean customPublisher = false;
+
+	/**
+	 * Construct a {@link LeaderInitiator} with a default candidate.
+	 *
+	 * @param client     Hazelcast client
+	 */
+	public LeaderInitiator(HazelcastInstance client) {
+		this(client, new DefaultCandidate());
+	}
+
+	/**
+	 * Construct a {@link LeaderInitiator}.
+	 *
+	 * @param client     Hazelcast client
+	 * @param candidate  leadership election candidate
+	 */
+	public LeaderInitiator(HazelcastInstance client, Candidate candidate) {
+		this.client = client;
+		this.candidate = candidate;
+	}
+
+	@Override
+	public void setApplicationEventPublisher(ApplicationEventPublisher applicationEventPublisher) {
+		if (!this.customPublisher) {
+			this.leaderEventPublisher = new DefaultLeaderEventPublisher(applicationEventPublisher);
+		}
+	}
+
+	/**
+	 * Start the registration of the {@link #candidate} for leader election.
+	 */
+	@Override
+	public synchronized void start() {
+		if (!this.running) {
+			this.mapLocks = this.client.getMap("spring-cloud-leader");
```

Comment:
Yeah... We aren't in a `could` here any more. Although still Spring :smile: 

Anyway I think it would be better to expose property to let end-users to choose a target Hazelcast Map.
But, right, with something like `spring-leader` by default would enough.